glslc/src/main.cc

```diff
@@ -97,11 +97,16 @@ bool GetOptionArgument(int argc, char** argv, int* index,
 }  // anonymous namespace
 
 int main(int argc, char** argv) {
-  std::vector<std::pair<std::string, EShLanguage>> input_files;
-  EShLanguage force_shader_stage = EShLangCount;
+  std::vector<std::pair<std::string, shaderc_shader_kind>> input_files;
+  shaderc_shader_kind forced_shader_stage = shaderc_glsl_infer_from_source;
   glslc::FileCompiler compiler;
   bool success = true;
   bool has_stdin_input = false;
+  // Contains all the macro names data. Current libshaderc interface requires
+  // null-terminated string, we need to 'pull out' and copy the macro names and
+  // hold their data here.
+  std::vector<std::string> macro_names;
+  std::vector<std::string> macro_values;
```

Comment:
Sadly, this means that `shaderc_compile_options_add_macro_definition` is not fulfilling the contract described in its blurb.  It should be copying the macro strings.